src/lib/InstantSearch.ts

```diff
@@ -256,21 +281,25 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
     };
 
     if (this._searchFunction) {
+      // this client isn't used to actually search, but required for the helper
+      // to not throw errors
+      const fakeClient = ({
+        search: () => new Promise(noop),
+      } as any) as Client;
+
       this._mainHelperSearch = mainHelper.search.bind(mainHelper);
       mainHelper.search = () => {
         const mainIndexHelper = this.mainIndex.getHelper();
         const searchFunctionHelper = algoliasearchHelper(
-          {
-            search: () => new Promise(noop),
-          },
-          mainIndexHelper.state.index,
-          mainIndexHelper.state
+          fakeClient,
+          mainIndexHelper!.state.index,
```

Comment:
We can remove the `!` if we put it after the `getHelper` call. It would be less redundant.

Reply:
If I move this, I still need the exclamation mark, where do you mean to move it exactly?

Reply:
https://github.com/algolia/instantsearch.js/pull/3984/files/cecc940c97b45bb614e42dedc870eac620b04a55#diff-1fc36c4ea042d2ecb5a498416e5e0357R292